### DISCUSSION TITLE
Call entity page via ESPHome Service

### DISCUSSION THREAD

**Discussion by Doughboy68** [Blackymas/NSPanel_HA_Blueprint#1016] [Q&A] [answered]
I can use the ESPHome printf service to do things like page home, page alarm etc. But how do I access a specific entity's detail page (the light page for an RGB light for example)?

**Comment by edwardtfn** [comment, marked as answer]
### v3.x:

1. Open the related light (or cover, etc.) with a button. Take a look at the content of `sensor.xxxxx_entity_settings`.
2. Copy that json-like string.
3. Use that string to call the service `esphome.xxxxx_set_entity_settings'.
* Take a look on the format of that string and you might be able to create it yourself without the need of steps 1 and 2 in case your don't have a button associated to the light you want see.

### v4.x (`dev`):

Call the service `esphome.xxxxx_open_entity_settings_page`.
Parameters:
- page: The page to open, between `climate`, `light`, `fan` or `cover`
- page_label: The title of the page
- page_icon: The icon to be shown on the top-left corner (Those are the icons available: https://htmlpreview.github.io/?https://github.com/jobr99/Generate-HASP-Fonts/blob/master/cheatsheet.html)
- page_icon_color: A 3-integer list for Red, Green and Blue (like `[255, 0, 0]` for a red icon). 
- entity: The `entity_id` to be controlled.
- back_page: The page to be shown when the close icon (the X in the top-right corner) is pressed. Usually is `home`.

**Reply by edwardtfn** [reply]
For Alarm page (only available from v4.0) call service `esphome.xxxxx_alarm_settings`.

In that case the parameters are similar to the `settings_page` or coming directly from the entity's attributes. The exception is `mui_alarm` which is a list of translated strings to be shown for buttons. Something like: `["Home", "Away", "Vacations", "Night", "Custom bypass", "Disarm"]` (hopefully my memory came with the correct order 😆 ).

**Comment by Doughboy68** [comment]
Exactly what I wanted.
Wanted to make a button so opens fan settings page with one press.
Used a script in HA and blueprint assigned to button calls it.
Only drawback is doesn't provide icon update on home page for button status.

**Comment by joztoz** [comment]
Hello, please how to open climate page, because now when I open it, it show me slider, but not active, is different as I have when I click on temp on home page, how to call to open exactly same climate page as its for setting heating?
Thank you

**Reply by edwardtfn** [reply]
How are you opening the page when you get the incomplete page?

**Reply by joztoz** [reply]
hellom, I have tried various entities and pages, but it always show inactive slider:

service: esphome.bytbavchod_open_entity_settings_page
data:
  page: climate
  page_label: climate
  page_icon: E651
  page_icon_color:
    - 255
    - 0
    - 0
  back_page: home
  entity: climate.bytbavchod_thermostat

**Reply by joztoz** [reply]
I dont know what I did, but changed entity to esphome.bytbavchod and for one time it works, but now it show again unactive slider.

**Reply by edwardtfn** [reply]
Could you please go to Developer Tools, tab States, find this entity `climate.bytbavchod_thermostat` and share here it's state and attributes?

**Reply by joztoz** [reply]
sure, thank you for helping:)
state: heat
atributes:
hvac_modes: off, heat
min_temp: 16
max_temp: 30
target_temp_step: 0.5
preset_modes: home, Off
current_temperature: 29.5
temperature: 16
hvac_action: idle
preset_mode: none
friendly_name: bytbavchod Thermostat
supported_features: 17

**Reply by edwardtfn** [reply]
I don't see anything wrong, but will try to duplicate this tomorrow and come back to you.
By the way, which integration is this?

**Reply by joztoz** [reply]
its dev, I guess 

ESPhome Version | 2023.8.3 Sep 27 2023, 20:15:47

**Reply by edwardtfn** [reply]
And which TFT are you using? EU, US or US Landscape?

**Reply by edwardtfn** [reply]
About integration, I mean which integration is your thermostat?

**Reply by joztoz** [reply]
sorry dont understand how you mean which integration?
its EU version and is in home assistant.
I just want to have possibility, that if I click on right button, it will jump to climate page.

**Reply by joztoz** [reply]
now I see when I increased timeout page, it will work, in way that after approx 30s I am able to click on fire icon and it will show active slider, than I can choose degree, but it will not read previous value, but it will start from min temp, than if I close and click on degrees on home page, here is seted value from call climate page.

**Reply by joztoz** [reply]
ok now I definitelly found answer, in entity need to be embedded_climate
 that full code is:
 service: esphome.bytbavchod_open_entity_settings_page
data:
  page: climate
  page_label: ""
  page_icon: ""
  page_icon_color:
    - 255
    - 0
    - 0
  back_page: home
  entity: embedded_climate

**Reply by edwardtfn** [reply]
Ok, but then this is for when you are using the climate add-on. It will control everything locally.
